front/src/common/IntervalsEditor/IntervalsEditorCommonForm.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
+import { fixLinearMetadataItems, resizeSegment } from 'common/IntervalsDataViz/data';
+import { notEmpty } from 'common/IntervalsDataViz/utils';
+import { IntervalItem } from './types';
+
+type IntervalsEditorFormProps = {
+  data: IntervalItem[];
+  interval: IntervalItem;
+  selectedIntervalIndex: number;
+  setData: (newData: IntervalItem[]) => void;
+  totalLength: number;
+};
+
+const IntervalsEditorCommonForm = ({
+  data,
+  interval,
+  selectedIntervalIndex,
+  setData,
+  totalLength,
+}: IntervalsEditorFormProps) => {
+  const { t } = useTranslation();
+
+  /**
+   * Resize segment from inputs
+   */
+  // TODO: fix bug
+  const resizeSegmentByInput = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    context: 'begin' | 'end'
+  ) => {
+    const gap = parseFloat(e.target.value) - interval[context];
+    const result = resizeSegment(data, selectedIntervalIndex, gap, context, false);
+    const fixedResults = fixLinearMetadataItems(result.result?.filter(notEmpty), totalLength);
+    setData(fixedResults);
+  };
+
+  return (
+    <div>
+      <InputSNCF
+        type="number"
+        id="item-begin"
+        label={t('begin')}
```

Comment:
Could'nt find the json file for this component. Does the translation works properly if it's only "hard written" it this file ?

Reply:
Done, thanks ! It was not translated before :)